src/algebra/group_action_hom.lean

```diff
@@ -0,0 +1,259 @@
+* `A →+[M] B` is `distrib_mul_action_hom M X Y`.
+* `R →+*[M] S` is `mul_semiring_action_hom M X Y`.
+
+-/
+
+universes u v w u₁
+
+variables (M : Type u) [monoid M]
+variables (X : Type v) [mul_action M X]
+variables (Y : Type w) [mul_action M Y]
+variables (Z : Type u₁) [mul_action M Z]
+variables (A : Type v) [add_monoid A] [distrib_mul_action M A]
+variables (A' : Type v) [add_group A'] [distrib_mul_action M A']
+variables (B : Type w) [add_monoid B] [distrib_mul_action M B]
+variables (B' : Type w) [add_group B'] [distrib_mul_action M B']
+variables (C : Type u₁) [add_monoid C] [distrib_mul_action M C]
+variables (R : Type v) [semiring R] [mul_semiring_action M R]
+variables (R' : Type v) [ring R'] [mul_semiring_action M R']
+variables (S : Type w) [semiring S] [mul_semiring_action M S]
+variables (S' : Type w) [ring S'] [mul_semiring_action M S']
+variables (T : Type u₁) [semiring T] [mul_semiring_action M T]
+variables (G : Type u) [group G] (H : set G) [is_subgroup H]
+
+set_option old_structure_cmd true
+
+/-- Equivariant functions. -/
+@[nolint has_inhabited_instance]
+structure mul_action_hom :=
+(to_fun : X → Y)
+(map_smul' : ∀ (m : M) (x : X), to_fun (m • x) = m • to_fun x)
+
+notation X ` →[`:25 M:25 `] `:0 Y:0 := mul_action_hom M X Y
+
+namespace mul_action_hom
+
+instance : has_coe_to_fun (X →[M] Y) :=
+⟨_, λ c, c.to_fun⟩
+
+variables {M X Y}
+
+@[simp] lemma map_smul (f : X →[M] Y) (m : M) (x : X) : f (m • x) = m • f x :=
+f.map_smul' m x
+
+@[ext] theorem ext : ∀ {f g : X →[M] Y}, (∀ x, f x = g x) → f = g
+| ⟨f, _⟩ ⟨g, _⟩ H := by { congr' 1, ext x, exact H x }
+
+theorem ext_iff {f g : X →[M] Y} : f = g ↔ ∀ x, f x = g x :=
+⟨λ H x, by rw H, ext⟩
+
+variables (M) {X}
+
+/-- The identity map is equivariant. -/
+protected def id : X →[M] X :=
+⟨id, λ _ _, rfl⟩
+
+@[simp] lemma id_apply (x : X) : mul_action_hom.id M x = x := rfl
+
+variables {M X Y Z}
+
+/-- Composition of two equivariant maps. -/
+def comp (g : Y →[M] Z) (f : X →[M] Y) : X →[M] Z :=
+⟨g ∘ f, λ m x, calc
+g (f (m • x)) = g (m • f x) : by rw f.map_smul
+          ... = m • g (f x) : g.map_smul _ _⟩
+
+@[simp] lemma comp_apply (g : Y →[M] Z) (f : X →[M] Y) (x : X) : g.comp f x = g (f x) := rfl
+
+@[simp] lemma id_comp (f : X →[M] Y) : (mul_action_hom.id M).comp f = f :=
+ext $ λ x, by rw [comp_apply, id_apply]
+
+@[simp] lemma comp_id (f : X →[M] Y) : f.comp (mul_action_hom.id M) = f :=
+ext $ λ x, by rw [comp_apply, id_apply]
+
+local attribute [instance] mul_action.regular
+
+variables {G} (H)
+
+/-- The canonical map to the left cosets. -/
+def to_quotient : G →[G] quotient_group.quotient H :=
+⟨coe, λ g x, rfl⟩
+
+@[simp] lemma to_quotient_apply (g : G) : to_quotient H g = g := rfl
+
+end mul_action_hom
+
+/-- Equivariant additive monoid homomorphisms. -/
+@[nolint has_inhabited_instance]
+structure distrib_mul_action_hom extends A →[M] B, A →+ B.
+
+/-- Reinterpret an equivariant additive monoid homomorphism as an additive monoid homomorphism. -/
```

Comment:
I don't understand that docstring and the line below it.

Reply:
This is the make lint happy by adding a docstring to the automatically generated `distrib_mul_action_hom.to_add_monoid_hom`.

cf. https://github.com/leanprover-community/mathlib/blob/c8c18697d4019608b756edf7989290303a96071a/src/topology/algebra/open_subgroup.lean#L26-L27